handler.go

```diff
@@ -244,11 +252,42 @@ func (h *Handler) renderPage(w http.ResponseWriter, r *http.Request, path string
 		page.PreloadModules = template.HTML(h.manifest.GeneratePreloadModules(chunk.Src))
 	}
 
+	var tmplName string
+	if path == "/" {
+		tmplName = "index.html"
+	} else {
+		tmplName = path
+	}
+
 	// Find the template to use.
-	tmpl, ok := h.templates[path]
+	tmpl, ok := h.templates[tmplName]
+	if !ok {
+		// Try alternative names
```

Comment:
Maybe we can add a comment about what this actually does? I mean, yes, anyone can scan it. But it takes a few seconds to read, and a comment just makes it obvious with a glimpse. We could even extract this to an individual method (but this seems optional to me). What do you think?

Reply:
That's good. What do you think of something like this?

```go
	// Find the template by name.
	tmpl, ok := h.templates[tmplName]

	// Catch common variations. If a template isn't found by the exact name,
	// check for variations like: "page", "page.html", or "/page.html", to match
	// how users might have registered the template.
	if !ok {
		variations := []string{
			strings.TrimPrefix(tmplName, "/"),
			strings.TrimPrefix(tmplName, "/") + ".html",
			strings.TrimSuffix(strings.TrimPrefix(tmplName, "/"), ".html"),
			tmplName + ".html",
		}
		for _, variant := range variations {
			if t, found := h.templates[variant]; found {
				tmpl = t
				ok = true
				break
			}
		}
	}
```